### PR TITLE
Fix compilation without deprecated APIs in OpenSSL 1.1

### PR DESCRIPTION
**Pull request by neheb**
Prior commit was tested with 1.0.2. This one with 1.1.

This is for 1.7.x

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -370,7 +370,7 @@ if test "x$SSL" != "xno"; then
 				#include <openssl/ssl.h>
 				#include <openssl/dh.h>
 			]], [[
-				SSL_CTX* ctx = SSL_CTX_new(TLSv1_method());
+				SSL_CTX* ctx = SSL_CTX_new(SSLv23_method());
```
[the remark this answers is not in the document]
Comment (neheb) [reply configure.ac]: OpenSSL 1.1 deprecates TLS_vX. This results in the configure script not finding OpenSSL.

The replacement is just TLS. But that will not work with 1.0.2. SSLv23 works with both.

File: include/znc/Utils.h [hunk 2]
```diff
@@ -219,6 +219,10 @@ class CTable : protected std::vector<std::vector<CString>> {
 #include <openssl/aes.h>
 #include <openssl/blowfish.h>
 #include <openssl/md5.h>
+#if OPENSSL_VERSION_NUMBER < 0x10100000L
```
[the remark this answers is not in the document]
Comment (neheb) [reply include/znc/Utils.h]: Fixed.

**Comment by DarthGandalf** [comment]
Hi @neheb , I see this:
`/znc-src/src/Utils.cpp: In static member function 'static void CUtils::GenerateCert(FILE*, const CString&)':
/znc-src/src/Utils.cpp:102:52: error: 'X509_getm_notBefore' was not declared in this scope`

https://travis-ci.org/znc/znc/jobs/459997299

Please take a look

**Comment by neheb** [comment]
I would guess that buildbot is using LibreSSL. Define needs to be updated.

**Pull request merged**